pyam/utils.py

```diff
@@ -132,6 +132,24 @@ def format_data(df, **kwargs):
     if isinstance(df, pd.Series):
         df = df.to_frame()
 
+    # Check for R-style year columns, converting where necessary
+    def convert_r_columns(c):
+        try:
+            first = c[0]
+            second = c[1:]
+            if first == 'X':
+                try:
+                    #  bingo! was X2015 R-style, return the integer
+                    return int(second)
+                except:
+                    # nope, not an int, fall down to final return statement
+                    pass
+        except:
+            # not a string/iterable/etc, fall down to final return statement
+            pass
+        return c
+    df.columns = map(convert_r_columns, df.columns)
```

Comment:
```suggestion
    df.columns = df.columns.map(convert_r_columns)
```

may be more pandas like?